src/DynamoCore/Models/AnnotationModel.cs

```diff
@@ -0,0 +1,379 @@
+            set
+            {
+                textBlockHeight = value;
+                Top = InitialTop - textBlockHeight;
+                Height = InitialHeight + textBlockHeight;
+            }
+        }
+
+        private double fontSize = 10;
+        public Double FontSize
+        {
+            get { return fontSize; }
+            set
+            {
+                fontSize = value;
+                RaisePropertyChanged("FontSize");
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AnnotationModel"/> class.
+        /// </summary>
+        /// <param name="nodes">The nodes.</param>
+        /// <param name="notes">The notes.</param>
+        /// <param name="loadFromGraph">This is true when graph is loaded from XML</param>
+        public AnnotationModel(IEnumerable<NodeModel> nodes, IEnumerable<NoteModel> notes, bool loadFromGraph=false)
+        {                                 
+            var nodeModels = nodes as NodeModel[] ?? nodes.ToArray();           
+            var noteModels = notes as NoteModel[] ?? notes.ToArray();
+
+            this.SelectedModels = nodeModels.Concat(noteModels.Cast<ModelBase>()).ToList();
+
+            if (!loadFromGraph)
+                CreateGroupingOnModels();
+        }
+
+
+        private void model_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case "X":
+                    CreateGroupingOnModels();
+                    break;
+                case "Y":
+                    CreateGroupingOnModels();
+                    break;
+                case "Position":
+                    CreateGroupingOnModels();
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Recalculate the group when a node is disposed
+        /// </summary>
+        /// <param name="node">The node.</param>
+        private void node_Disposed(NodeModel node)
+        {
+            var nodesList = this.SelectedModels.ToList();
+            bool remove = nodesList.Remove(node);
+            if (remove)
+            {
+                SelectedModels = nodesList;
+                CreateGroupingOnModels();
+            }
+        }
+
+        /// <summary>
+        /// Recaluclate the group when a note is disposed
+        /// </summary>
+        /// <param name="note">The note.</param>
+        private void note_Disposed(NoteModel note)
+        {
+            var notesList = this.SelectedModels.ToList();
+            bool remove = notesList.Remove(note);
+            if (remove)
+            {
+                SelectedModels = notesList;
+                CreateGroupingOnModels();
+            }
+        }
+
+        /// <summary>
+        /// Creates the grouping on selected models.
+        /// </summary>      
+        private void CreateGroupingOnModels()
```

Comment:
How about `UpdateBoundaryFromSelection`?  This sounds like its a constructor or something...